### DISCUSSION TITLE
Best way to implement multiple auth ( User, Admin )?

### DISCUSSION THREAD

**Discussion by Jarrodsz** [sergiodxa/remix-auth#200] [Q&A]
What would be the best way to implement Multiple Auth?
I want to implement a safe way to login the admin separate from a regular User. 
so in my application im Already using a separate Admin and User model.

Would I create 2 Auth services 1 for Admin and 1 for User, or would it be safe to just have 1 single app/services/auth.service.tsx with separate methods for admin and user?

**Comment by sergiodxa** [comment]
Since you have two tables on your DB, I would use two Authenticator instances with their own sessionStorage (or a single sessionStorage but with a different sessionKey).

Then in the admin routes I would use the admin authenticator and in other routes I would use the user authenticator.
